receiver/dockerstatsreceiver/logs_receiver.go

```diff
@@ -0,0 +1,201 @@
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/docker/docker/api/types/events"
+	"github.com/docker/docker/api/types/filters"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/receiver"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/docker"
+)
+
+const dockerEventPrefix = "docker"
+
+type logsReceiver struct {
+	config   *Config
+	settings receiver.Settings
+
+	cancel      context.CancelFunc
+	consumer    consumer.Logs
+	eventPoller *dockerEventPoller
+}
+
+func newLogsReceiver(set receiver.Settings, config *Config, consumer consumer.Logs) *logsReceiver {
+	return &logsReceiver{
+		config:   config,
+		settings: set,
+		consumer: consumer,
+	}
+}
+
+func (r *logsReceiver) Start(ctx context.Context, _ component.Host) error {
+	var err error
+	client, err := docker.NewDockerClient(&r.config.Config, r.settings.Logger)
+	if err != nil {
+		return err
+	}
+
+	if err = client.LoadContainerList(ctx); err != nil {
+		return err
+	}
+
+	cctx, cancel := context.WithCancel(ctx)
+	r.cancel = cancel
+	r.eventPoller = newDockerEventPoller(r.config, client, r.settings.Logger, r.consumeDockerEvent)
+	go r.eventPoller.Start(cctx)
+	return nil
+}
+
+func getDockerBackoffConfig(config *Config) *backoff.ExponentialBackOff {
+	b := &backoff.ExponentialBackOff{
+		InitialInterval:     config.MinDockerRetryWait,
+		MaxInterval:         config.MaxDockerRetryWait,
+		MaxElapsedTime:      0,
+		Multiplier:          1.5,
+		RandomizationFactor: 0.5,
+		Clock:               backoff.SystemClock,
+	}
+	b.Reset()
+	return b
+}
+
+// dockerEventPoller manages the lifecycle and event stream of the Docker daemon connection.
+type dockerEventPoller struct {
+	config       *Config
+	client       *docker.Client
+	logger       *zap.Logger
+	eventHandler func(context.Context, *events.Message) error
+	backoff      *backoff.ExponentialBackOff
+}
+
+func newDockerEventPoller(
+	config *Config,
+	client *docker.Client,
+	logger *zap.Logger,
+	handler func(context.Context, *events.Message) error) *dockerEventPoller {
+	return &dockerEventPoller{
+		config:       config,
+		client:       client,
+		logger:       logger,
+		eventHandler: handler,
+		backoff:      getDockerBackoffConfig(config),
+	}
+}
+
+func (d *dockerEventPoller) Start(ctx context.Context) {
+	for {
+		filterArgs := filters.NewArgs()
+		for k, v := range d.config.Logs.Filters {
+			for _, elem := range v {
+				filterArgs.Add(k, elem)
+			}
+		}
+		// event stream can be interrupted by async errors (connection or other).
+		// client caller must retry to restart processing. retry with backoff here
+		// except for context cancellation.
+		eventChan, errChan := d.client.Events(ctx, events.ListOptions{
```

Comment:
What is the minimum Docker API version supported for these operations? If it is a higher version than otherwise documented in this receiver, it should be noted in the docs